.github/workflows/build-emsdk.yml

```diff
@@ -1,27 +1,87 @@
-name: build-emsdk
+name: Build WASM (Emscripten)
 
-on: [push]
+# Run CI only on changes to main branch, or any PR to main. Do not run CI on 
+# any other branch. Also, skip any non-source changes from running on CI
+on:
+  push:
+    branches: main
+    paths-ignore:
+      - 'docs/**'
+      - 'examples/**'
+      - '.gitignore'
+      - '*.rst'
+      - '*.md'
+      - '.github/workflows/*.yml'
+      # re-include current file to not be excluded
+      - '!.github/workflows/build-emsdk.yml'
+
+  pull_request:
+    branches: main
+    paths-ignore:
+      - 'docs/**'
+      - 'examples/**'
+      - '.gitignore'
+      - '*.rst'
+      - '*.md'
+      - '.github/workflows/*.yml'
+      # re-include current file to not be excluded
+      - '!.github/workflows/build-emsdk.yml'
 
 jobs:
   build:
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
+    env:
+      # pin SDK version to the latest, update manually
+      SDK_VERSION: 0.2.0
+      SDK_ARCHIVE: python-wasm-sdk-stable.tar.bz2
+
+      # use the most recent cython from github, but pin on a commit for CI
+      # stability. This is also needed to benefit from caching cython installs
+      LATEST_CYTHON_COMMIT: 8e29b6d47f6f5b10ec1a37f06db440156ac2ac2e
+
+      WHEELHOUSE_CYTHON: /tmp/wheelhouse/cython
+
     steps:
     - uses: actions/checkout@v3.0.2
-    - name: pygame-wasm-builder prepare
+
+    - name: Cache Cython
+      id: cache-cython
+      uses: actions/cache@v3
+      with:
+        path: ${{ env.WHEELHOUSE_CYTHON }}
+        key: wasm-ubuntu-cython-${{ env.LATEST_CYTHON_COMMIT }}-path-${{ env.WHEELHOUSE_CYTHON }}
+
+    # This builds the cython wheel and stores it in cache too
+    - name: Download and build cython on cache miss
+      if: steps.cache-cython.outputs.cache-hit != 'true'
+      run: |
+        mkdir -p $WHEELHOUSE_CYTHON
+        pip wheel --wheel-dir $WHEELHOUSE_CYTHON git+https://github.com/cython/cython.git@$LATEST_CYTHON_COMMIT
+
+    - name: Install latest cython and regen
+      run: |
+        pip install --no-index --find-links $WHEELHOUSE_CYTHON --pre cython
+        touch $(find | grep pxd$)
+        python3 setup.py cython_only
+
+    - name: Install python-wasm-sdk
+      run: |
+        mkdir python-wasm-sdk && cd python-wasm-sdk
+        curl -sL --retry 5 https://github.com/pygame-web/python-wasm-sdk/releases/download/$SDK_VERSION/$SDK_ARCHIVE | tar xvPj
```

Comment:
maybe pipe that to `| wc -l` to clean up output

Reply:
I would prefer more verbose output because it would make debugging easier when something goes wrong

Reply:
i think curl or bzip2 or tar  would send something meaningfull on stderr

Reply:
Probably, but I was thinking about the kind of errors where some file/directory is missing, it would be handy to be able to inspect what exactly was un-archived if needed.